x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/forcemerge.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiDescribedFormGroup,
+  EuiFieldNumber,
+  EuiSpacer,
+  EuiSwitch,
+  EuiTextColor,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { LearnMoreLink } from './learn_more_link';
+import { ErrableFormRow } from './form_errors';
+import { Phases, PhaseWithForcemergeAction } from '../../../../../common/types';
+import { PhaseValidationErrors, propertyof } from '../../../services/policies/policy_validation';
+
+const forcemergeLabel = i18n.translate('xpack.indexLifecycleMgmt.warmPhase.forceMergeDataLabel', {
+  defaultMessage: 'Force merge data',
+});
+
+interface Props<T extends PhaseWithForcemergeAction> {
```

Comment:
It is not really clear why we need the generic type `T` here - `PhaseWithForcemergeAction` should cover all the props we will use in this component. I might be missing something though?

I would remove the generic and just have `phaseData: PhaseWithForcemergeAction` below.